src/plugins/region_map/public/region_map_visualization.js

```diff
@@ -90,29 +93,60 @@ export function createRegionMapVisualization({
       this._kibanaMap.useUiStateFromVisualization(this._vis);
     }
 
+    async _loadConfig(fileLayerConfig) {
+      // Load the selected layer from the metadata-service.
+      // Do not use the selectedLayer from the visState.
+      // These settings are stored in the URL and can be used to inject dirty display content.
+
+      if (fileLayerConfig.isEMS) {
+        //Hosted by EMS. Metadata needs to be resolved through EMS.
+        return await serviceSettings.loadFileLayerConfig(fileLayerConfig);
+      }
+
+      if (fileLayerConfig.layerId && fileLayerConfig.layerId.startsWith(`${ORIGIN.EMS}.`)) {
+        //fallback for older saved objects
+        return await serviceSettings.loadFileLayerConfig(fileLayerConfig);
```

Comment:
The conditions on lines `101` and `106` both result in:

```
return await serviceSettings.loadFileLayerConfig(fileLayerConfig);
```

Should this be a different return value or should these conditions be consolidated into a single `if` statement with the same return value?